soc/arm/nuvoton_npcx/common/soc_gpio.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _NUVOTON_NPCX_SOC_GPIO_H_
+#define _NUVOTON_NPCX_SOC_GPIO_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Pin number for each GPIO device */
+#define NPCX_GPIO_PORT_PIN_NUM 8U
+
+/**
+ * @brief Get GPIO device instance by port index
+ *
+ * @param port GPIO device index
+ *
+ * @retval Pointer to structure device
+ * @retval NULL Invalid parameter of GPIO port index
+ */
+struct device *soc_get_gpio_dev(int port);
```

Comment:
I have thought of using device_get_binding() to solve it. But we need to maintain a gpio device name array in miwu driver or using sprintf() to generate gpio device name string to get device address dynamically with port parameter. They might not be better than the current solution. That's all I can figure out so far.

Reply:
Right, but I am wondering why you need the port parameter. Typically if you have a device for each port, then you can get the device from the DT and the idea of a port number is no-longer userful.

Anyway if you can't figure it out, let's leave it as is for now.

Reply:
Regarding why we need the 'port' parameter, this is because the original Zephyr gpio callback mechanism restores the gpio callback structures in the heap directly. Please refer https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/gpio.h#L363 and https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/basic/button/src/main.c#L41. Since they are declared and registered by gpio_add_callback() in the application layer, the npcx miwu driver has no chance to know how many gpios needs the MIWU's support during compiling. (Please notice a gpio event triggers related WUI signal and invokes MIWU's ISR.) 

Hence, we handle gpio callback functions in miwu driver directly. By borrowing unused 24 bits of pin_mask which is part of gpio_callback structure and already exists in heap, the gpio driver saved the necessary information in manage_callback() api function. It includes:
1. Related WUI signal of target gpio which is used for setting its MIWU device configuration.
2. Port number of this gpio which is used for getting its gpio device address during invoking its callback functions in MIWU's ISR. Since there is only unused 24 bits space, we cannot save the device address directly. Saving port number is an alternative solution.

Reply:
OK I see. I am not enough of an expert to know if there is a better way, so it seems OK to me.